digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/AddressLinkBuilder.scala

```diff
@@ -109,19 +109,30 @@ trait AddressLinkBuilder {
   }
 
 
+  /**
+    * Fuses roadaddresses that do not contain history. We will later have to refactor this so that we check if history is fusable and figure out rules for fusing.
+    * @param roadAddresses roadaddress seq
+    * @return  fused roadaddress seq
+    */
+
   def fuseRoadAddress(roadAddresses: Seq[RoadAddress]): Seq[RoadAddress] = {
+    val oldRoadAddresses = roadAddresses.filterNot(x=>x.id==(-1000))
```

Comment:
this should be after next if clause so we dont have to do unnecessary extra filter.
then in else call we could do some partitioning to have both new and old road addresses at once like for e.g. val (newAddresses, oldAddresses) = roadAddresses.partition(_.id==(-1000))